### ISSUE TITLE
10.0.5 breaks with Typescript and ||= operator

### ISSUE THREAD

**Issue by vpontis** [vercel/next.js#20920]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

v15.3.0

**What browser are you using?**

n/a

**What operating system are you using?**

MacOS

**How are you deploying your application?**

n/a

**Describe the Bug**

The `||=` operator causes a babel issue when running `yarn dev` in a Next.js repo.

**Expected Behavior**

It should work

**To Reproduce**

Upgrade to Next.js 10.0.5 and add a line with the `||=` operator in Typescript. 

With next.js 10.0.5, we can no longer use the `||=` operator in Typescript. 

I have prepared an [example repo here](https://github.com/vpontis/next-js-ts-issue--) which only slightly modifies the `yarn create next-app` repo by adding a line with `||=`

This might not be an issue on Linux.

```
➜  test-ts-assignment git:(main) yarn dev
yarn run v1.22.10
$ next dev
ready - started server on http://localhost:3000
event - compiled successfully
event - build page: /
wait  - compiling...
event - build page: /next/dist/pages/_error
error - ./pages/index.tsx 7:6
Module parse failed: Unexpected token (7:6)
File was processed with these loaders:
 * ./node_modules/next/dist/build/webpack/loaders/next-babel-loader.js
You may need an additional loader to handle the result of these loaders.
| export default function Home() {
|   let x = 'hi';
>   x ||= 'hello';
|   console.log(x);
|   return /*#__PURE__*/_jsxDEV("div", {
```

**Comment by caribou-code** [comment]
I've pulled the example repo and ran locally. Seems to be working as expected, so perhaps just a local issue for you @vpontis?

**Comment by UgRoss** [comment]
I've tried to reproduce that but it's working for me as well.

<img width="497" alt="CleanShot 2021-01-24 at 00 17 38@2x" src="https://user-images.githubusercontent.com/5703885/105615472-a4da7a00-5dd9-11eb-8976-fc940faac874.png">

`MacOS  11.2 / Node.js v14.15.1 `

As a suggestion, maybe try to reinstall modules and check if the error still happens.

**Comment by vpontis** [comment]
Hmmm, I ran into it a few times and some team members ran into it... Let me try again :)

**Comment by vpontis** [comment]
Hmm I still get this issue even after re-installing `node_modules` and it's not isolated to my computer. 

I'll see if I can figure out if anything fixes it...

![image](https://user-images.githubusercontent.com/1319079/105618546-527b6800-5d9d-11eb-9665-9b1fbf34ea0e.png)

**Comment by offfffz** [comment]
I can confirm this issue. The following are the steps I took.

1. clone the repo
2. run `yarn install`
3. run `yarn dev`
4. open http://localhost:3000 in the web browser.
5. got the following error
    <img width="586" alt="Screen Shot 2021-01-24 at 18 41 34" src="https://user-images.githubusercontent.com/4332800/105629137-cbe58a00-5e73-11eb-8f32-9efeb8e58260.png">

My environment was: 
- Macbook Air M1 (not sure if it is related)
- nvm 0.37.2 
- node 15.5.1
- yarn 1.22.10

**Comment by caribou-code** [comment]
I can confirm this is something to do with Node version (or a side effect of that).

The following environment **does** work:

- Node 12.20.0
- npm 6.14.8
- Yarn 1.22.10

The following environment **does not** work:

- Node 15.3.0
- npm 7.0.14
- Yarn 1.22.10

Notice the yarn version remains consistent, so this is to do with the Node/npm version. To avoid any doubt in my tests, I set the relevant Node version before doing a fresh repo clone, install and run for each test.

I did wonder whether this had something to do with npm 7 (which ships as default with Node 15) installing peer dependencies by default (which it apparently does). That could potentially mean an outdated sub-dependency version being used for one of the sub-dependencies, causing this error. However if yarn doesn't use npm behind the scenes, that won't be the case.

**Comment by UgRoss** [comment]
Yeah, tried to install Node.js v15 and the issue appeared.

✅ Works: `MacOS 11.2 / Node.js v14.15.1 / Yarn 1.22.10`
❌ Doesn't work: `MacOS 11.2 / Node.js v15.6.0 / Yarn 1.22.10`

**Comment by switz** [comment]
Yup noticed this when upgrading to 10.0.5.

**Comment by kachkaev** [comment]
Same problem here on

- Next.js 1.11.2 / 12.0.4
- Node.js 16.13.0 LTS

Code (inside a `.tsx` file):

```ts
  let x: string | undefined = undefined;
  x ??= "hello";

  return <div>{x}</div>;
```

**Comment by balazsorban44** [comment]
Hi, I confirmed this now works on Next 12, both with SWC and Babel. If you still have problems, please open a new issue with a reproduction, using the latest `canary` release. Thanks!
